gum/arch-arm64/gumarm64writer.c

```diff
@@ -1992,15 +1993,37 @@ gum_arm64_writer_commit_literals (GumArm64Writer * self)
     if (r->width != GUM_LITERAL_64BIT)
       continue;
 
+    /*
+     * Whilst instructions in aarch64 are always in little endian (even on
+     * big-endian systems), the data is in native endian. Thus since we wish to
+     * support writing code for big-endian systems on little-endian targets and
+     * vice versa, we need to check the writer configuration.
+     */
```

Comment:
Let's move this comment to where the `data_endian` field is defined. We can then avoid duplicating it further down.

Reply:
Done.